tests/Dialect/ModArith/Transforms/find_mac.mlir

```diff
@@ -0,0 +1,23 @@
+// RUN: heir-opt --arith-to-mod-arith --mod-arith-to-mac %s | FileCheck %s --enable-var-scope
+
+// CHECK-LABEL: @double_mac
+// CHECK-SAME: (%[[ARG:.*]]: !Z32767_i16_) -> [[T:.*]] {
+func.func @double_mac(%arg0: i16) -> i16 {
+  %c1 = arith.constant 1: i16
+  %c2 = arith.constant 2 : i16
+  %c3 = arith.constant 3: i16
+
+  // CHECK: %[[MAC1:.*]] = mod_arith.mac %[[ARG]], %{{.*}}, %{{.*}} : [[T]]
+  // CHECK: %[[ADD:.*]] = mod_arith.add %[[MAC1]], %{{.*}} : [[T]]
+  // CHECK: %[[MAC2:.*]] = mod_arith.mac %[[ADD]], %{{.*}}, %{{.*}} : [[T]]
+  // CHECK: return %[[MAC2:.*]] : [[T]]
+
```

Comment:
filecheck test here should assign names to the operands and make sure they are the expected ones :)